addon/components/file-browser/component.js

```diff
@@ -0,0 +1,23 @@
+import Ember from 'ember';
+import layout from './template';
+
+/* use:
+ * {{file-browser rootNode=node selectedFile=file
+ *   onClickFile=(action) onClickNode=(action)}}
+ */
+export default Ember.Component.extend({
+    layout,
+    classNames: ['file-browser'],
+
+    selectedPath: Ember.computed('selectedFile', function() {
+        let file = this.get('selectedFile');
+        if (!file) {
+            return null;
+        }
+        let pathArray = this.get('selectedFile.materializedPath').split('/');
```

Comment:
Also, clicking on a subfolder should not send an MFR request - the download url will be undefined.  
![screen shot 2016-06-17 at 2 22 40 pm](https://cloud.githubusercontent.com/assets/9755598/16160575/fd7a4dd2-3496-11e6-8fe1-e4fa9665f416.png)